var/spack/repos/builtin/packages/arbor/package.py

```diff
@@ -48,7 +50,7 @@ class Arbor(CMakePackage, CudaPackage):
     extends('python', when='+python')
     depends_on('python@3.6:', when="+python", type=('build', 'run'))
     depends_on('py-numpy', when='+python', type=('build', 'run'))
-    depends_on('py-pybind11@2.6:', when='+python', type=('build', 'run'))
+    depends_on('py-pybind11@2.8.1:', when='+python', type=('build', 'run'))
```

Comment:
did you mean to change this for all previous versions too?

Reply:
No, good catch! The switchover occurred started v0.5.3-dev, so I think that means I should replace the line with the following:

```
    depends_on('py-pybind11@2.6:', when='+python,@:0.5.2', type=('build', 'run'))
    depends_on('py-pybind11@2.8.1:', when='+python,@0.5.3:', type=('build', 'run'))
```
I'm not 100% sure though I can indeed comma separate constraints on different packages/properties like this. Could you confirm this?

Reply:
how about

```suggestion
    with when('+python'):
        depends_on('py-pybind11@2.6:', type=('build', 'run'))
        depends_on('py-pybind11@2.8.1:', when='@0.5.3:', type=('build', 'run'))
```